docs/csharp/language-reference/operators/snippets/SwitchExpressions.cs

```diff
@@ -0,0 +1,137 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace operators
+{
+    public enum Directions
+    {
+        Up,
+        Down,
+        Right,
+        Left
+    }
+
+    public enum Orientation
+    {
+        North,
+        South,
+        East,
+        West
+    }
+
+    public static class SwitchExpressions
+    {
+        public static void Examples()
+        {
+            InitialExample();
+
+            var collection = new int[]{1,2,3,4,5,6,7,8,9};
+            TypeExample(collection);
+            RecursiveExample(collection);
+            RecursiveExample(collection[0..0]);
+            RecursiveExample(collection[0..1]);
+            RecursiveExample(collection[0..2]);
+            CaseGuardExample(collection.AsEnumerable());
+            CaseGuardExample(collection[0..0].AsEnumerable());
+            CaseGuardExample(collection[0..1].AsEnumerable());
+            CaseGuardExample(collection[0..2].AsEnumerable());
+            try {
+                ExhaustiveExample(default(List<int>));
+            } catch (ArgumentNullException e)
+            {
+                Console.WriteLine($"Caught expected exception: {e.Message}");
+            }
+
+        }
+
+        private static void InitialExample()
+        {
+            // <SnippetBasicStructure>
+            var direction = Directions.Right;
+            Console.WriteLine($"Map view direction is {direction}");
+
+            var orientation = direction switch
+            {
+                Directions.Up    => Orientation.North,
+                Directions.Right => Orientation.East,
+                Directions.Down  => Orientation.South,
+                Directions.Left  => Orientation.West,
+            };
+            Console.WriteLine($"Cardinal orientation is {orientation}");
+            // </SnippetBasicStructure>
+        }
+
+        private static void TypeExample<T>(IEnumerable<T> sequence)
+        {
+            // <SnippetTypePattern>
+            var third = sequence switch
+            {
+                System.Array array => array.GetValue(2),
+                IList<T> list => list[2],
+                IEnumerable<T> seq => seq.Skip(2).First(),
+            };
+            Console.WriteLine(third);
+            // </SnippetTypePattern>
+        }
+
+        private static void RecursiveExample<T>(IEnumerable<T> sequence)
+        {
+            // <SnippetRecursivePattern>
+            var third = sequence switch
```

Comment:
nit-pick: should be named differently because, in the case of short arrays, the result is not third. However, if expression-bodied, the variable name is not concern anymore.